src/main/java/com/ita/if103java/ims/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,217 @@
+package com.ita.if103java.ims.dao;
+
+import com.ita.if103java.ims.entity.User;
+import com.ita.if103java.ims.exception.CRUDException;
+import com.ita.if103java.ims.exception.EntityNotFoundException;
+import com.ita.if103java.ims.mapper.UserRowMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+
+    private UserRowMapper userRowMapper;
+    private JdbcTemplate jdbcTemplate;
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+    private static Logger logger = LoggerFactory.getLogger(UserDaoImpl.class);
+
+    @Autowired
+    public UserDaoImpl(DataSource dataSource, UserRowMapper userRowMapper) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.bCryptPasswordEncoder = new BCryptPasswordEncoder();
+        this.userRowMapper = userRowMapper;
+    }
+
+    private PreparedStatement createStatement(User user, String encryptedPassword, LocalDateTime localDateTime,
+                                              String emailUUID, Connection connection) throws SQLException {
+        PreparedStatement preparedStatement = connection.prepareStatement(Queries.SQL_CREATE_USER, Statement.RETURN_GENERATED_KEYS);
+
+        preparedStatement.setString(1, user.getFirstName());
+        preparedStatement.setString(2, user.getLastName());
+        preparedStatement.setString(3, user.getEmail());
+        preparedStatement.setString(4, encryptedPassword);
+        preparedStatement.setObject(5, user.getRole());
+        preparedStatement.setObject(6, localDateTime);
+        preparedStatement.setObject(7, localDateTime);
+        preparedStatement.setBoolean(8, user.isActive());
+        preparedStatement.setString(9, emailUUID);
```

Comment:
this is 'magic numbers'. It is hard to maintain, better to declare some variable and use here increment, ie:
`int i=1;
preparedStatement.setString(i++, user.getFirstName());
preparedStatement.setString(i++, user.getLastName());`